src/basic/fileio.c

```diff
@@ -1354,6 +1354,27 @@ int fputs_with_separator(FILE *f, const char *s, const char *separator, bool *sp
         return 0;
 }
 
+int fputs_with_newline(FILE *f, const char *s) {
+        assert(s);
```

Comment:
doesn't matter, but you could even make this a NOP if s is specified NULL or empty.